### PR TITLE
Initialize Terminal Preview settings from Terminal settings

### LINKED ISSUE #6855: [discussion] Initialize Terminal Preview settings from Terminal settings
If Terminal Preview is installed for the first time, can it initialize settings from Terminal settings?

Usecase is trying out Preview with same settings as regular version. Currently it's reset to default settings.

In any case, this is not a big issue

### PR DESCRIPTION
**Pull request by huiyooumich**
Windows Terminal Preview gets existing settings from Release build if
Preview settings are empty

This ensures that when settings are empty or not existent, we check if
we're currently in a preview build and if we are, we attempt to grab
settings from the Release build's setting path instead. We tested it
manually by changing settings in Release build and confirming that
changes migrated to Preview when settings are empty or not existent.
Additionally, we tested that settings.json of the running build changed.
We also ran existing TAEF testing locally and it passed.

In LoadAll() function in
src\cascadia\TerminalSettingsModel\CascadiaSettingsSerialization.cpp, we
first checked if the settings file us empty/exists via settingsString.
If it does not and we are in the Preview build, we try loading the
Release build's settings. We created modified versions of
CascadiaSettings::_settingsPath() and GetBaseSettingsPath() to get the
path for the Release build's settings. If the Release build settings do
exist and firstTimeSetup is true, we set it to settingsString so it can
be written to disk via WriteSettingsToDisk(). Note that currently we
hardcode the path of the Release build. This pull request was worked on
with @Dannihu01.

## Validation Steps Performed Test1: Setting to firstTimeSetup is true
and loading settings.json from WT release when release exists -> Result:
settings.json AND GUI reflected WT release’s settings

Test2: Setting to firstTimeSetup is true and loading settings.json from
WT release when release doesn’t exist -> Result: settings.json AND GUI
reflected DEFAULT settings

Test3: (After running Test1) Setting to firstTimeSetup is false and
seeing if current settings.json matches WT release. (See if it doesn’t
change) -> Result: settings.json AND GUI reflected WT release’s settings

Closes #6855 

Co-authored-by: Danniell Hu <dannihu@umich.edu>

### CODE REVIEW

File: src/cascadia/TerminalSettingsModel/FileUtils.cpp [hunk 1]
```diff
@@ -43,6 +44,33 @@ namespace winrt::Microsoft::Terminal::Settings::Model
...
+            // Returns a path like C:\Users\<username>\AppData\Local
+            std::filesystem::path parentDirectoryForSettingsFile{ localAppDataFolder.get() };
+
+            //Appending \Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState to the settings path
+            parentDirectoryForSettingsFile /= ReleaseSettingsFolder;
+
+            if (!IsPackaged())
+            {
+                parentDirectoryForSettingsFile /= UnpackagedSettingsFolderName;
+            }
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsModel/FileUtils.cpp:67]: I'm not sure this is right for unpackaged versions of the Terminal. I might want to tap in @DHowett for this bit, hes a bit more familiar with running the Terminal unpackaged. 

Just looking at my own files, I've only got a `%localAppData%\Microsoft\Windows Terminal`.

Heck, I'm not sure this is ever relevant for Preview builds running unpackaged - I think they always just use `%localAppData%\Microsoft\Windows Terminal` anyways. I might be wrong here.

File: src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp [hunk 2]
```diff
@@ -713,8 +713,28 @@ void SettingsLoader::_executeGenerator(const IDynamicProfileGenerator& generator
...
+    auto settingsString = ReadUTF8FileIfExists(_settingsPath()).value_or(std::string{});
+    auto firstTimeSetup = settingsString.empty();
+
+    // If it's the firstTimeSetup and a preview build, then try to
+    // read settings.json from the Release stable file path if it exists.
+    // Otherwise use default settings file provided from original settings file
+    bool releaseSettingExists = false;
+    if (firstTimeSetup)
+    {
+#if defined(WT_BRANDING_PREVIEW)
```
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp:725]: Clever, simple, easy to parse what's happening. 👍

**Review by zadjii-msft — COMMENTED** [review]
Okay, I think this is pretty straightforward. I'd ✅, but I'm a little curious about how this works unpackaged, esp with Preview branding, so I want a second opinion.

**Comment by zadjii-msft** [comment]
(sorry for the delays in reviewing 😅)

**Review by lhecker — APPROVED** [review]
I believe this change is fine. With unpackaged WT, I don't believe we make a distinction for the settings path between Preview and non-Preview.

**Comment by zadjii-msft** [comment]
So, for unpackaged "preview" branded builds, it'll just happen to work anyways? Like, this is never an issue, unpackaged terminal just always works this way. Even if you ran an unpackaged Preview build first, with no existing settings, it would never find the "release" settings, because it would look in the nonsense path `%localAppData%\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\Microsoft\Windows Terminal` for the settings file, definitely not find it, and continue on its merry way? 

Okay I suppose I'm fine with this, but I might follow up with a cleanup PR with comments to that effect.

**Review by zadjii-msft — APPROVED** [review]
as noted. Thanks!

**Comment by carlos-zamora** [comment]
@msftbot make sure @DHowett signs off on this

**Comment by zadjii-msft** [comment]
Hey @DHowett review this so this can be in 1.15.

File: src/cascadia/TerminalSettingsModel/FileUtils.cpp [hunk 3]
```diff
@@ -43,6 +44,33 @@ namespace winrt::Microsoft::Terminal::Settings::Model
...
 
+    // Returns a path like C:\Users\<username>\AppData\Local\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState
+    // to the path of the stable release settings
+    std::filesystem::path GetReleaseSettingsPath()
+    {
+        static std::filesystem::path baseSettingsPath = []() {
+            wil::unique_cotaskmem_string localAppDataFolder;
+            // KF_FLAG_FORCE_APP_DATA_REDIRECTION, when engaged, causes SHGet... to return
+            // the new AppModel paths (Packages/xxx/RoamingState, etc.) for standard path requests.
+            // Using this flag allows us to avoid Windows.Storage.ApplicationData completely.
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/FileUtils.cpp:55]:
```suggestion
            // We're using KF_FLAG_NO_PACKAGE_REDIRECTION to ensure that we always get the
            // user's actual local AppData directory.
```

**Review by DHowett — APPROVED** [review]
Thank you for doing this! Sorry I let it sit so long!

**Review by Fulya2133 — APPROVED** [review]

**Review by Fulya2133 — APPROVED** [review]
function (request, socket, head) { }

**Pull request merged**